graphql/database/users/__tests__/deleteUser.test.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-env jest */
+
+import uuid from 'uuid/v4'
+import moment from 'moment'
+
+import UserModel from '../UserModel'
+import deleteUser from '../deleteUser'
+import {
+  DatabaseOperation,
+  getMockUserContext,
+  getMockUserInfo,
+  getMockUserInstance,
+  mockDate,
+  setMockDBResponse,
+  clearAllMockDBResponses,
+} from '../../test-utils'
+import constructFullWidget from '../../widgets/constructFullWidget'
+import BaseWidgetModel from '../../widgets/baseWidget/BaseWidgetModel'
+import UserWidgetModel from '../../widgets/userWidget/UserWidgetModel'
+import getWidgets from '../../widgets/getWidgets'
+
+jest.mock('../../databaseClient')
+jest.mock('../../widgets/getWidgets')
+jest.mock('uuid/v4')
+
+const defaultUserContext = getMockUserContext()
+
+beforeAll(() => {
+  mockDate.on()
+})
+
+afterAll(() => {
+  mockDate.off()
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+  clearAllMockDBResponses()
+})
+
+describe('deleteUser scrubs desired fields: email, username and widgets', () => {
+  it('', async () => {
+    const userInfo = getMockUserInfo()
+    expect.assertions(3)
+
+    // Set mock query responses.
+    const userWidgetsToGet = [
+      new UserWidgetModel({
+        userId: userInfo.id,
+        widgetId: 'abc',
+      }),
+      new UserWidgetModel({
+        userId: userInfo.id,
+        widgetId: 'def',
+      }),
+    ]
+    const baseWidgetsToGet = [
+      new BaseWidgetModel({
+        id: 'abc',
+        position: 2,
+      }),
+      new BaseWidgetModel({
+        id: 'def',
+        position: 1,
+      }),
+    ]
+    const sortedFullWidgets = [
+      constructFullWidget(userWidgetsToGet[1], baseWidgetsToGet[1]),
+      constructFullWidget(userWidgetsToGet[0], baseWidgetsToGet[0]),
+    ]
+    getWidgets.mockResolvedValueOnce(sortedFullWidgets)
+
+    const uuidVal = 'abcd'
+    uuid.mockReturnValueOnce(uuidVal)
+
+    // Mock database responses.
+    const userReturnedFromCreate = getMockUserInstance(
+      Object.assign({}, userInfo)
+    )
+    setMockDBResponse(DatabaseOperation.UPDATE, {
+      Attributes: userReturnedFromCreate,
+    })
+
+    const updateMethod = jest.spyOn(UserModel, 'update')
+    const userWidgetUpdateMethod = jest.spyOn(UserWidgetModel, 'update')
+
+    await deleteUser(defaultUserContext, userInfo.id)
+
+    userWidgetsToGet.forEach(widget =>
+      expect(userWidgetUpdateMethod).toHaveBeenCalledWith(defaultUserContext, {
+        userId: userInfo.id,
+        widgetId: widget.widgetId,
+        enabled: false,
+        data: {},
+        config: {},
+        updated: moment.utc().toISOString(),
+      })
+    )
+
+    expect(updateMethod).toHaveBeenCalledWith(defaultUserContext, {
+      id: userInfo.id,
+      email: `deleted-${uuidVal}@example.com`,
+      username: `deleted-${uuidVal}`,
+      backgroundImage: UserModel.fieldDefaults.backgroundImage(),
+      deleted: true,
+      updated: moment.utc().toISOString(),
+    })
+  })
```

Comment:
Once you've added the user claim `auth_time` check, add a couple of tests for it.